src/components/SearchModal/CurrencySearch.tsx

```diff
@@ -103,12 +103,12 @@ export function CurrencySearch({
     return Object.values(allTokens).filter(getTokenFilter(debouncedQuery))
   }, [allTokens, debouncedQuery])
 
-  const balances = useAllTokenBalances()
+  const [balances, balancesIsLoading] = useAllTokenBalances()
   const sortedTokens: Token[] = useMemo(() => {
     return filteredTokens.sort(tokenComparator.bind(null, balances))
   }, [balances, filteredTokens])
 
-  const filteredSortedTokens = useSortTokensByQuery(debouncedQuery, sortedTokens)
+  const filteredSortedTokens = useSortTokensByQuery(debouncedQuery, sortedTokens, balancesIsLoading)
```

Comment:
Instead of adding `balancesIsLoading` to `useSortTokensByQuery`, where it is not explicitly relevant, add it to the `useMemo` deps array on L109, so that `balancesIsLoading` is used where it is referenced, and doesn't leak state out of this function.

Reply:
sounds good, made changes from our discussion on slack.

Reply:
Sounds good, done